backend/api/routes/users.js

```diff
@@ -46,18 +46,41 @@ router.route("/get-all").get(authenticateMiddleware, async (req, res) => {
       throw new Error("Error fetching the current user", userErr);
     }
 
-    //TODO: ONCE SAVED CONNECTIONS IS IMPLEMENTED (THE PAGE) we need to filter out and only display streamers the user has not connected with. Dont want to display current connections on the discover page.
-    //fetch all profile rows
-    const { data: streamers, streamersError } = await supabase
+    //get the connections this user has. We get connections from connections table and select the connected streamer id whenever that connection's user_id matches ours
+    const { data: connections } = await supabase
+      .from("connections")
+      .select("connected_streamer_id")
+      .eq("user_id", userId);
+
+    //just to make our life easier map through connections cause each itme in it is a database object. just extract the id and put it in a new array.
+    const connectedStreamersIds = connections.map(
+      (c) => c.connected_streamer_id
+    );
+
+    //here we arent calling query yet just building it and storing it in a variable
+    let query = supabase
       .from("profiles")
       .select("*")
       //neq means select all profiles that arent equal to current user session id. We dont want oursevles showing up in recommendation
       .neq("id", userId)
       //ascending false allows us to get newly created streamers first
       .order("created_at", { ascending: false });
 
-    if (streamersError) {
-      throw new Error("Error fetching the streamers", streamersError);
+    //if user had connections then we want to filter them out
+    if (connectedStreamersIds.length > 0) {
+      //do this by using the not query. Got some weird looking syntaxt cause the third argument it takes is a string in this exact format (abc, efg, 123)
+      //Then the not means we dont want any of thoses connectedStreamersIds that are in our id column in profiles tables
+      query = query.not("id", "in", `(${connectedStreamersIds.join(",")})`);
```

Comment:
If is empty this results will be the same, right? I think you can directly nest this. Also, I remember that you can nest supabase queries, something like:
```javascript
  .not('id', 'in', supabase
    .from('connections')
    .select('connected_streamer_id')
    .eq("user_id", userId)
  )
```
Not sure if there's any performance benefit there but just in case that you want to put everything together